### ISSUE TITLE
queryInterface.bulkUpdate fails with MySQL

### ISSUE THREAD

**Issue by sebikeller** [sequelize/sequelize#10625]
## What are you doing?
Update a row in a table using only the QueryInterface, since I'm not able to not to use the Models directly.

```js
await sequelizeInstance.getQueryInterface().bulkUpdate('Table', { data: 'foo' }, { key: 'bar' }, { transaction });
```

## What do you expect to happen?
The row should be updated.

## What is actually happening?
An Exception is thrown:

```
TypeError: valueSets.map is not a function
    at Function.bulkBuild (node_modules/sequelize/lib/model.js:1986:22)
    at Function._target.(anonymous function) [as bulkBuild] (node_modules/sequelize-typescript/lib/models/BaseModel.js:36:41)
    at Query.handleSelectQuery (node_modules/sequelize/lib/dialects/abstract/query.js:303:27)
    at Query.formatResults (node_modules/sequelize/lib/dialects/mysql/query.js:135:21)
    at Utils.Promise.then.then.results (node_modules/sequelize/lib/dialects/mysql/query.js:88:29)
    at tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:694:18)
    at _drainQueueStep (node_modules/bluebird/js/release/async.js:138:12)
    at _drainQueue (node_modules/bluebird/js/release/async.js:131:9)
    at Async._drainQueues (node_modules/bluebird/js/release/async.js:147:5)
    at Immediate.Async.drainQueues [as _onImmediate] (node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:705:18)
    at tryOnImmediate (timers.js:676:5)
    at processImmediate (timers.js:658:5)
```

The problem seems to be, that the `sequelizeInstance.getQueryInterface().bulkUpdate()` function does not set the `options.type` to `connection.QueryTypes.BULKUPDATE` or `connection.QueryTypes.UPDATE`. So the result parser thinks it's a SELECT query.

__Dialect:__ mysql
__Sequelize version:__ 4.43
__Tested with latest release:__ No (If yes, specify that version)

**Comment by delikatessa** [comment]
adding `{type: Sequelize.QueryTypes.UPDATE}` as `options` param fixed this for me

**Comment by sebikeller** [comment]
As I have written in the problem report, but with other methods like `bulkInsert` the correct type is added when calling the method.

**Comment by krazi3** [comment]
@papb I have added a fix for this issue. 
This is my first PR submitted to sequelize and i would like to continue contributing more in the future. 😄

**Comment by papb** [comment]
@krazi3 Thank you very much! :grimacing: This is great to hear. I'll see you around then! Feel free to ping me (@papb) if you need anything.

**Comment by papb** [comment]
Fixed by #11515
